### PR TITLE
adjust padding when removed out of fixed layer

### PR DESCRIPTION
**Pull request by sriramkrish85**

### CODE REVIEW

File: src/service/fixed-layer.js [hunk 1]
```diff
@@ -295,6 +295,17 @@ export class FixedLayer {
   removeFixedElement_(element) {
     for (let i = 0; i < this.fixedElements_.length; i++) {
       if (this.fixedElements_[i].element == element) {
+        this.vsync_.run({
```
Comment (dvoytenko) [inline src/service/fixed-layer.js]: As discussed, just `element.style.top = ''` should be enough, but we may also need `element.style.zIndex = ''` - please check.

Comment (sriramkrish85) [inline src/service/fixed-layer.js]: Done - 

ZIndex is not needed its being handled in returnFromFixedLayer_ (which is where it is needed)

File: extensions/amp-sidebar/0.1/amp-sidebar.js [hunk 2]
```diff
@@ -183,8 +183,8 @@ export class AmpSidebar extends AMP.BaseElement {
       this.closeMask_();
       this.element.removeAttribute('open');
       this.element.setAttribute('aria-hidden', 'true');
-      this.viewport_.removeFromFixedLayer(this.element);
       timer.delay(() => {
+        this.viewport_.removeFromFixedLayer(this.element);
```
Comment (dvoytenko) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: I assume also when only `!this.isOpen_()`?

File: src/service/fixed-layer.js [hunk 3]
```diff
@@ -295,6 +295,9 @@ export class FixedLayer {
   removeFixedElement_(element) {
     for (let i = 0; i < this.fixedElements_.length; i++) {
       if (this.fixedElements_[i].element == element) {
+        this.vsync_.mutate(()=> {
```
Comment (dvoytenko) [inline src/service/fixed-layer.js]: nit: space

File: test/functional/test-fixed-layer.js [hunk 4]
```diff
@@ -376,6 +376,28 @@ describe('FixedLayer', () => {
...
+
+      // Remove.
+      fixedLayer.vsync_ = {
+        mutate: function(callback) {
+          callback();
+        },
+      };
+      fixedLayer.removeElement(element3);
+      expect(fixedLayer.fixedElements_).to.have.length(2);
+      expect(element3.style.top).to.equal('');
```
Comment (dvoytenko) [inline test/functional/test-fixed-layer.js]: We should check if the `top != ''` first :) Or, alternatively, we can fill it in.

**Comment by dvoytenko** [comment]
getting there. just a few more comments.

File: test/functional/test-fixed-layer.js (hunk 4 above)
Comment (sriramkrish85) [inline test/functional/test-fixed-layer.js]: Done

File: src/service/fixed-layer.js (hunk 3 above)
Comment (sriramkrish85) [inline src/service/fixed-layer.js]: Done

File: extensions/amp-sidebar/0.1/amp-sidebar.js (hunk 2 above)
Comment (sriramkrish85) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: yes, good catch, thanks!

**Comment by sriramkrish85** [comment]
PTAL

**Comment by dvoytenko** [comment]
LGTM

**Comment by sriramkrish85** [comment]
Fixes #2859

**Pull request merged**